debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/FindModuleByNameCommand.java

```diff
@@ -0,0 +1,109 @@
+import com.ibm.j9ddr.vm29.j9.ObjectModel;
+import com.ibm.j9ddr.vm29.j9.Pool;
+import com.ibm.j9ddr.vm29.j9.SlotIterator;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ModulePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9PoolPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.JavaVersionHelper;
+
+/**
+ * FindModuleByName command find the modules corresponding to its module name
+ * 
+ * Example:
+ * 			!findmodulebyname java.base
+ * Example output: 
+ * 			Searching for modules named 'java.base' in VM=13053677e00
+ *			java.base			!j9module 0x00000130550C7E88
+ *			Found 1 module(s) named java.base
+ */
+public class FindModuleByNameCommand extends Command
+{
+	public FindModuleByNameCommand()
+	{
+		addCommand("findmodulebyname", "<moduleName>", "find the modules corresponding to its module name");
+	}
+
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException
+	{
+		if (args.length == 0) {
+			printUsage(out);
+			return;
+		}
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			if (JavaVersionHelper.ensureJava9AndUp(vm, out)) {
+				J9PoolPointer pool = vm.modularityPool();
+				Pool<J9ModulePointer> modulePool = Pool.fromJ9Pool(pool, J9ModulePointer.class);
+				SlotIterator<J9ModulePointer> poolIterator = modulePool.iterator();
+				J9ModulePointer modulePtr = null;
+
+				int hitCount = 0;
+				String searchModuleName = args[0];
+				J9ObjectPointer moduleNameObjPtr = null;
+				PatternString pattern = new PatternString(searchModuleName);
+
+				out.println(String.format("Searching for modules named '%s' in VM=%s", searchModuleName,
+						Long.toHexString(vm.getAddress())));
+				while (poolIterator.hasNext()) {
+					modulePtr = poolIterator.next();
+					moduleNameObjPtr = modulePtr.moduleName();
+					if (ObjectModel.isPointerInHeap(vm, moduleNameObjPtr)
+							&& pattern.isMatch(J9ObjectHelper.stringValue(moduleNameObjPtr))) {
+						hitCount++;
+
+						String moduleName = J9ObjectHelper.stringValue(moduleNameObjPtr);
+						String hexAddress = modulePtr.getHexAddress();
+						out.printf("%-30s !j9module %s%n", moduleName, hexAddress);
+					}
+				}
+				out.println(String.format("Found %d module(s) named '%s'", hitCount, searchModuleName));
+			}
+		} catch (CorruptDataException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+	}
+
+	/**
+	 * Prints the usage for the findmodulebyname command.
+	 *
+	 * @param out The PrintStream the usage statement prints to
+	 */
+	private void printUsage(PrintStream out) 
+	{
+		out.println("findmodulebyname <moduleNamePattern> - find the module corresponding to its given name pattern (like 'java.base')");
```

Comment:
The help can be improved, I suggest:

findmodulebyname <moduleNamePattern> - find the modules corresponding to a name pattern (e.g. 'java.*')

Notice it makes it clear that more than one module might be found and shows the form of pattern used (I assume  'java.*' is the same as the Java pattern 'java\..*').